VENTokenField/VENBackspaceTextField.m

```diff
@@ -24,15 +24,34 @@
 
 @implementation VENBackspaceTextField
 
-- (BOOL)keyboardInputShouldDelete:(UITextField *)textField
-{
-    if (self.text.length == 0) {
-        if ([self.delegate respondsToSelector:@selector(textFieldDidEnterBackspace:)]) {
-            [self.delegate textFieldDidEnterBackspace:self];
+- (BOOL)keyboardInputShouldDelete:(UITextField *)textField {
+    BOOL shouldDelete = YES;
+
+    if ([UITextField instancesRespondToSelector:_cmd]) {
+        BOOL (*keyboardInputShouldDelete)(id, SEL, UITextField *) = (BOOL (*)(id, SEL, UITextField *))[UITextField instanceMethodForSelector:_cmd];
```

Comment:
It is sort of voodoo, to quote an answer on stackoverflow: 
> There's usually no reason to need IMPs unless you're doing serious voodoo--is there something specific you want to do?

The `instanceMethodForSelector` is a method of `NSObject` and returns `the address of the receiver’s implementation of a method so it can be called as a function (IMP)`. In this case the address of the implementation method is retrieved and run in order to  and result is set to `keyboardInputShouldDelete`.
It would be _a bit_ easier to read if the implementation method reference was assigned to a IMP variable first, and then set the BOOL based on calling that one [but still this is voodoo...].